librosa/onset.py

```diff
@@ -345,7 +345,7 @@ def onset_strength(
         **kwargs,
     )
 
-    return odf_all[0]
+    return np.squeeze(odf_all)
```

Comment:
This seems dangerous to me.  `np.squeeze` removes all singleton dimensions, where the old code was only removing one dimension.

Would `return odf_all[..., 0, :]` work instead?